plugins/catalog-react/src/components/EntityOwnerPicker/EntityOwnerPicker.tsx

```diff
@@ -35,111 +29,79 @@ import { Autocomplete } from '@material-ui/lab';
 import React, { useEffect, useMemo, useState } from 'react';
 import { useEntityList } from '../../hooks/useEntityListProvider';
 import { EntityOwnerFilter } from '../../filters';
-import { getEntityRelations } from '../../utils';
-import useAsync from 'react-use/lib/useAsync';
-import { errorApiRef, useApi } from '@backstage/core-plugin-api';
+import { useApi } from '@backstage/core-plugin-api';
 import { catalogApiRef } from '../../api';
-import { humanizeEntity, humanizeEntityRef } from '../EntityRefLink/humanize';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+import { useDebouncedEffect } from '@react-hookz/web';
+import PersonIcon from '@material-ui/icons/Person';
+import GroupIcon from '@material-ui/icons/Group';
+import { humanizeEntity } from '../EntityRefLink/humanize';
 
 /** @public */
 export type CatalogReactEntityOwnerPickerClassKey = 'input';
 
-const useStyles = makeStyles(
-  {
-    input: {},
-  },
-  {
-    name: 'CatalogReactEntityOwnerPicker',
-  },
-);
-
 const icon = <CheckBoxOutlineBlankIcon fontSize="small" />;
 const checkedIcon = <CheckBoxIcon fontSize="small" />;
 
 /** @public */
 export const EntityOwnerPicker = () => {
-  const classes = useStyles();
   const {
     updateFilters,
-    backendEntities,
     filters,
     queryParameters: { owners: ownersParameter },
   } = useEntityList();
+
   const catalogApi = useApi(catalogApiRef);
-  const errorApi = useApi(errorApiRef);
+  const [text, setText] = useState('');
+
+  const [{ value, loading }, handleFetch] = useAsyncFn(
+    async (request: { text: string } | { cursor: string; prev: Entity[] }) => {
+      const initialRequest = request as { text: string };
+      const cursorRequest = request as { cursor: string; prev: Entity[] };
+      const limit = 20;
+
+      if (cursorRequest.cursor) {
+        const response = await catalogApi.queryEntities({
+          cursor: cursorRequest.cursor,
+          limit,
+        });
+        return {
+          ...response,
+          items: [...cursorRequest.prev, ...response.items],
+        };
+      }
+
+      return catalogApi.queryEntities({
+        fullTextFilter: {
+          term: initialRequest.text || '',
+          fields: [
+            'metadata.name',
+            'kind',
+            'spec.profile.displayname',
```

Comment:
yeah in the search table the field are converted to lowercase, so `displayName` won't work. Probably the `queryEntities` endpoint should have handled this internally

Reply:
Ah, gotcha, I misread where this was coming from, assumed it was the top level `fields` and not `fullTextFilter.fields`. Ideally, the backend would handle the casing transformation. 

Also, do we want to use a `fields` filter as well? We're returning the full entities here.

Reply:
it should be fine to return the full entity in this case since we are returning a batch of 20 entities per request. I don't think it will improve anything

Reply:
Agree it won't be noticeable, but might still be worth restricting as adopters could have larger than expected User/Group entities and we only need a subset of fields.